miniml/compiler/compile.scm

```diff
@@ -2479,22 +2484,47 @@
        (cons block-count (cons const-count (+ 1 block-count)))
        (cons const-count (cons (+ 1 const-count) block-count)))))
 
-(define (compile-type env name tdef)
+(define (expand-constr-def env type-name constr-def)
+  (match constr-def
+    ((constr-name . ('VariantTuple arity))
+     (cons env (cons constr-name arity)))
+    ((constr-name . ('VariantRecord record-def))
+     ; inline records are expanded naively:
+     ;     type t = A of { x : int }
+     ; becomes
+     ;     type t = A of t#A
+     ;     and t#A = { x : int }
```

Comment:
This is interesting, this means that the layout won't match OCaml's. I think it is fine, as inline records are not used for primitive functions though, but this is important to note.

Reply:
Indeed; I added a note to this effect. (It is not very hard to implement OCaml's layout, but it would be a more invasive change, so I would delay until we perceive an actual need.)